packages/wallet-sdk/src/core/provider/interface.ts

```diff
@@ -42,10 +42,47 @@ export interface AppMetadata {
   appChainIds: number[];
 }
 
-export interface Preference {
+type PostOnboardingAction = 'none' | 'onramp' | 'magicspend';
+
+type OnrampPrefillOptions = {
+  contractAddress?: string;
+  amount: string;
+  chainId: number;
+};
+
+export type Preference = {
   options: 'all' | 'smartWalletOnly' | 'eoaOnly';
+  /**
+   * @deprecated this is only for internal use
+   */
   keysUrl?: string;
```

Comment:
since this is only used by us, can we remove it for now

if anything, we will always catch the problem before we roll out as this is used internally i think?

Reply:
Might be good to list it as deprecated for a few versions just in case someone externally is using it

Reply:
hmm prob there won't be any one using it externally, let's remove it?

Reply:
Sounds good